### PR TITLE
[WebXR] Add detectedPlanes, improve detectedMeshes, add tests

### PR DESCRIPTION
**Pull request by sorskoot**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://immersive-web.github.io/real-world-geometry/plane-detection.html#plane-set
  - https://immersive-web.github.io/real-world-meshing/#mesh-set
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

---

Changes:
- Add detectPlanes to XRFrame interface
- Move detectMeshes to the correct location in the file, below comment about draft and together with related definitions
- Add optional initiateRoomCapture to XRSession
- Remove legacy plane detection definition
- Correct type of `lastChangeTime` on XRMesh
- Add optional `semanticLabel` attribute to XRMesj
- Add tests for these changes

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/webxr/webxr-tests.ts
1. types/webxr/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 557112600605789efb92e7947f22748eeee19d47
</p>

File: types/webxr/index.d.ts [hunk 1]
```diff
@@ -662,7 +653,22 @@ interface XRPlane {
...
+     * XRFrame is extended to contain detectedPlanes attribute which contains
+     * all planes that are still tracked in the frame.
+     *
+     * The set is initially empty and will be populated by the update planes
+     * algorithm. If this attribute is accessed when the frame is not active,
+     * the user agent MUST throw InvalidStateError.
+     *
+     * @see https://immersive-web.github.io/real-world-geometry/plane-detection.html#plane-set
+     */
+    readonly detectedPlanes: XRPlaneSet;
```
Comment (RaananW) [inline types/webxr/index.d.ts:671]: This object only exists if the plane detection feature is enabled. There is no way to define this condition in the declaration file. I think optional ("?") or " | undefined" is required here.

File: types/webxr/index.d.ts [hunk 2]
```diff
@@ -678,22 +684,34 @@ interface XRMesh {
...
+     * XRFrame is extended to contain detectedMeshes attribute
+     * which contains all meshes that are still tracked in the frame.
+     *
+     * The set is initially empty and will be populated by the update meshes algorithm.
+     * If this attribute is accessed when the frame is not active, the user agent
+     * MUST throw InvalidStateError.
+     *
+     * @see https://immersive-web.github.io/real-world-meshing/#mesh-set
+     */
+    readonly detectedMeshes: XRMeshSet;
```
Comment (RaananW) [inline types/webxr/index.d.ts:698]: Same comment as above

**Review by RaananW — CHANGES_REQUESTED** [review]

**Review by RaananW — APPROVED** [review]

**Comment by sorskoot** [comment]
Ready to merge

**Pull request merged**